### PR TITLE
Add 0.17.0 release notes

### PR DESCRIPTION
**Pull request by max-sixty**
Let me know any changes here, fairly soon if possible! 

And please make changes liberally — what are the important features we should highlight in the heading?

Thanks

### CODE REVIEW

**Comment by keewis** [comment]
I would probably include `quiver`, but otherwise looks good to me.

Am I correct in assuming that you will also try to fix the entries in `whats-new.rst` in this PR? If so, it might be good to add a entry for #4711 and to move the `drop_isel` entry to "new features".

**Comment by max-sixty** [comment]
Great, done!

File: doc/whats-new.rst [hunk 1]
```diff
@@ -105,6 +105,9 @@ New Features
 - Most rolling operations use significantly less memory. (:issue:`4325`).
   By `Deepak Cherian <https://github.com/dcherian>`_.
```
Comment (keewis) [inline doc/whats-new.rst:119]:
```suggestion
```

File: doc/whats-new.rst [hunk 2]
```diff
@@ -189,6 +189,8 @@ Documentation
 - Add concat examples and improve combining documentation (:issue:`4620`, :pull:`4645`).
   By `Ray Bell <https://github.com/raybellwaves>`_ and
   `Justus Magin <https://github.com/keewis>`_.
+- Added docs on vectorized indexing; (:pull:`4711`).
+  By `Eric Keenan <https://github.com/EricKeenan`_.
```
Comment (keewis) [inline doc/whats-new.rst:205]:
```suggestion
- Added docs on vectorized indexing (:pull:`4711`).
  By `Eric Keenan <https://github.com/EricKeenan>`_.
```

File: doc/whats-new.rst [hunk 3]
```diff
@@ -177,13 +194,26 @@ Bug fixes
...
+
+- ``dim`` argument to :py:meth:`DataArray.integrate` is being deprecated in
+  favour of a ``coord`` argument, for consistency with :py:meth:`Dataset.integrate`.
+  For now using ``dim`` issues a ``FutureWarning``. It will be removed in
+  version 0.19.0 (:pull:`3993`).
+  By `Tom Nicholas <https://github.com/TomNicholas>`_.
+- Deprecated ``autoclose`` kwargs from :py:func:`open_dataset` are removed; (:pull:`4725`).
+  By `Aureliana Barghini <https://github.com/aurghs>`_.
 - explicitly mention that :py:meth:`Dataset.update` updates inplace (:issue:`2951`, :pull:`4932`).
   By `Justus Magin <https://github.com/keewis>`_.
```
Comment (keewis) [inline doc/whats-new.rst:218]: we also have a `Deprecations` section above, could you merge those two?

Comment (keewis) [inline doc/whats-new.rst:218]: we also have a `Deprecations` section above, could you merge them?

**Comment by max-sixty** [comment]
@keewis thanks for fixing that up! 

Any final comments anyone? Any thoughts on the release "headline"?

**Review by dcherian — APPROVED** [review]
giant release!

thanks @keewis & @max-sixty

**Pull request merged**